feat: add a pop up to say that insuffcient balance

Angekarara opened the pull request:
Fixes #8

This PR implements error handling for insufficient balance scenarios by adding a popup notification. When a user attempts a transaction with insufficient funds, they will now receive clear feedback through a popup message, improving error handling and user communication.

Angekarara committed:
feat: add a pop up to say that insuffcient balance

toffee-k21 commented:
> error handling for insufficient balance scenarios by adding a popup notification. When a user attempts a transaction with insufficient funds, they will now receive clear feedback through a popup message, improving error handling and user communication.

Great job @Angekarara! 🎉 However, there's a required improvement — currently, the popup message is displayed statically (e.g., always showing "Not enough crypto"). Please update it to show dynamic messages based on different scenarios. Let me know if you need any help with this.

josue-igiraneza commented on hardhat/hardhat.config.js:
```diff
@@ -17,3 +17,6 @@ module.exports = {
     },
   },
 };
+
+console.log("RPC_URL_KEY:", process.env.RPC_URL_KEY);
+console.log("PRIVATE_KEY:", process.env.PRIVATE_KEY);
```
Are these console logs needed?

albert-byiringiro replied:
No, they are not a necessary.

josue-igiraneza commented on src/components/NFT.jsx:
```diff
@@ -1,118 +1,153 @@
+    const res = await contract.tokenURI(tokenId);
+    console.log("res", res);
+    const fetchedData = await fetch(res);
+    const jsonData = await fetchedData.json();
+    console.log("jsonData", typeof jsonData);
```
These ones too

albert-byiringiro replied:
No, they are not necessary.

toffee-k21 commented:
The changes in hardhat.config.js and deploy.js were actually not needed, as they weren't specifically requested in the issue either

albert-byiringiro committed:
refactor(auth): remove contract deployment script

albert-byiringiro committed:
refactor: remove unneccesary changes in the hardhat configuration file

albert-byiringiro committed:
refactor(ui): add functionality to show dynamic messages based on different scenarios

albert-byiringiro commented:
@toffee-k21, @Angekarara has allowed me to join the PR to add the improvements you suggested:

1. Revert changes made to hardhat.config.js
2. Remove the unnecessary file deploy.js
3. Remove unused imports, empty block statements and unclear comments
4. rename non-react properties to their real names
5. Update the popup to show dynamic messages for different scenarios

albert-byiringiro committed:
refactor: remove unnecessary console logs)

albert-byiringiro committed:
refactor: remove unused imports

albert-byiringiro committed:
refactor: changed attribute class to className

albert-byiringiro committed:
refactor: rename non-react properties in svg to real react properties

albert-byiringiro committed:
refactor: Empty block statement

albert-byiringiro committed:
remove unclear comments

Angekarara commented:
@toffee-k21 , Just following up on this PR. we’ve already addressed the requested changes a while back, so I wanted to check if there’s anything else you’d like me to do before it can be reviewed or merged.

Appreciate any updates when you get time! Thanks

toffee-k21 commented:
> @toffee-k21 , Just following up on this PR. we’ve already addressed the requested changes a while back, so I wanted to check if there’s anything else you’d like me to do before it can be reviewed or merged.
> 
> Appreciate any updates when you get time! Thanks

Hey @Angekarara, I think the useState for the alert might be unnecessary here. What's your take on it?

Angekarara commented:
> > @toffee-k21 , Just following up on this PR. we’ve already addressed the requested changes a while back, so I wanted to check if there’s anything else you’d like me to do before it can be reviewed or merged.
> > Appreciate any updates when you get time! Thanks
> 
> Hey @Angekarara, I think the useState for the alert might be unnecessary here. What's your take on it?

@toffee-k21 , Thank you for your review and pointing that out, i just realized it wasn't neccessary, i've removed it.

Angekarara committed:
fix: remove unnecessary alert state

toffee-k21 commented:
@Angekarara Awesome work! Everything looks good to me. Pulling it into my branch now. 🚀
